deps/build.jl

```diff
@@ -1,22 +1,40 @@
 #! /usr/bin/env julia
 
-pkg_dir = dirname(dirname(@__FILE__))
+path = joinpath(dirname(@__FILE__), "..", "src")
 
-@static if is_windows()
-    try
-      run(`where cl.exe`)
-    catch
-      error("ERROR: Could not find cl.exe.")
-    end
+if is_windows()
+    if success(`where cl.exe`)
+        info("Compiling with cl.exe")
+        run(`cl.exe /D_USRDLL /D_WINDLL /EHsc /Fo$(path) $(path)cclipper.cpp $(path)clipper.cpp /MT /link /DLL /OUT:$(path)cclipper.dll`)
+    else
+        import WinRPM
+        info("Compiling with WinRPM gcc-c++")
+        WinRPM.install("gcc-c++"; yes = true)
+        WinRPM.install("gcc"; yes = true)
+        WinRPM.install("headers"; yes = true)
+
+        gpp = Pkg.dir("WinRPM","deps","usr","x86_64-w64-mingw32","sys-root","mingw","bin","g++")
+        RPMbindir = Pkg.dir("WinRPM","deps","usr","x86_64-w64-mingw32","sys-root","mingw","bin")
+        incdir = Pkg.dir("WinRPM","deps","usr","x86_64-w64-mingw32","sys-root","mingw","include")
 
-    path = "$(pkg_dir)\\src\\"
-    run(`cl.exe /D_USRDLL /D_WINDLL /EHsc /Fo$(path) $(path)cclipper.cpp $(path)clipper.cpp /MT /link /DLL /OUT:$(path)cclipper.dll`)
+        push!(Base.Libdl.DL_LOAD_PATH, RPMbindir)
+        ENV["PATH"] = ENV["PATH"] * ";" * RPMbindir
```

Comment:
you shouldn't need to be (and generally should avoid) modifying path if you refer to the compiler by a full absolute path